packages/financial-templates-lib/src/helpers/acrossFeesCalculator.ts

```diff
@@ -0,0 +1,89 @@
+// This Util calculates the across realized LP fees. See https://gist.github.com/chrismaree/a713725e4fe96c531c42ed7b629d4a85
+// gist for a python implementation of the logic in this file.
+// TODO: add a link to chase's notebooks once they are production ready to explain how the logic works past the comments included here.
+
+import BN from "bn.js";
+import Decimal from "decimal.js";
+import Web3 from "web3";
+
+const { toBN, toWei, fromWei } = Web3.utils;
+const toBNWei = (number: string | number) => toBN(toWei(number.toString()).toString());
+const fixedPointAdjustment = toBNWei("1");
+
+export interface RateModel {
+  UBar: BN; // denote the utilization kink where the slope of the interest rate model changes.
+  R0: BN; // is the interest rate charged at 0 utilization
+  R1: BN; // R_0+R_1 is the interest rate charged at UBar
+  R2: BN; // R_0+R_1+R_2 is the interest rate charged at 100% utilization
+}
+
+// Calculate the rate for a 0 sized deposit (infinitesimally small).
+function calculateInstantaneousRate(rateModel: RateModel, utilization: BN) {
+  const piece1 = BN.min(utilization, rateModel.UBar).mul(rateModel.R1).div(rateModel.UBar);
+  const piece2 = BN.max(toBN("0"), utilization.sub(rateModel.UBar))
+    .mul(rateModel.R2)
+    .div(toBNWei("1").sub(rateModel.UBar));
+
+  return rateModel.R0.add(piece1).add(piece2);
+}
+
+//  Compute area under curve of the piece-wise linear rate model.
+function calculateAreaUnderRateCurve(rateModel: RateModel, utilization: BN) {
+  // Area under first piecewise component
+  const utilizationTilda1 = BN.min(utilization, rateModel.UBar);
```

Comment:
nit: why is it a "tilda" variable? Does that denote that its some delta? Some temporary variable?

Reply:
it's just a transient variable used before the kink. I renamed to be more descriptive.